File: fixtures/js/bind_all_interfaces/negative.ts
// SEC013-bind-all-interfaces: negative fixture
// None of these should produce SEC013 findings.

import express from 'express';
import http from 'http';

const app = express();

// OK: restricted to loopback.
app.listen("127.0.0.1", 3000);

// OK: port-only listen (no host string).
app.listen(3000);

// OK: not a bind callee.
console.log("0.0.0.0");

// OK: server listening on loopback.
const server = http.createServer(app);
server.listen("127.0.0.1:8080");

File: fixtures/python/bind_all_interfaces/negative.py
# SEC013-bind-all-interfaces: negative fixture
# None of these should produce SEC013 findings.

import socket
import flask
import uvicorn

# OK: restricted to loopback.
app = flask.Flask(__name__)
app.run("127.0.0.1", port=5000)

# OK: uvicorn with localhost host.
uvicorn.run(app, host="127.0.0.1", port=8000)

# OK: socket bound to loopback.
s = socket.socket()
s.bind(("127.0.0.1", 8080))

# OK: not a bind callee.
print("0.0.0.0")
